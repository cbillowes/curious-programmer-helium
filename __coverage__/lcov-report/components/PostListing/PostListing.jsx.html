<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/PostListing/PostListing.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/PostListing</a> PostListing.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React from "react"</span>
<span class="cstat-no" title="statement not covered" >import Metadata from "../Metadata/Post/Metadata"</span>
<span class="cstat-no" title="statement not covered" >import "./PostListing.scss"</span>
&nbsp;
class PostListing extends React.Component {
<span class="fstat-no" title="function not covered" >  getPostList</span>() {
    const postList = <span class="cstat-no" title="statement not covered" >[]</span>
<span class="cstat-no" title="statement not covered" >    this.props.postEdges.forEach(<span class="fstat-no" title="function not covered" >postEdge</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      postList.push({</span>
        path: postEdge.node.fields.slug,
        tags: postEdge.node.frontmatter.tags,
        cover: postEdge.node.frontmatter.cover,
        title: postEdge.node.frontmatter.title,
        date: postEdge.node.fields.date,
        excerpt: postEdge.node.excerpt,
        timeToRead: postEdge.node.timeToRead
      })
    })
<span class="cstat-no" title="statement not covered" >    return postList</span>
  }
<span class="fstat-no" title="function not covered" >  render</span>() {
    const list = <span class="cstat-no" title="statement not covered" >this.getPostList()</span>
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className="posts"&gt;
        {list.map(<span class="fstat-no" title="function not covered" >post</span> =&gt; {
<span class="cstat-no" title="statement not covered" >          return (</span>
            &lt;section key={`post-${post.path}`}&gt;
              &lt;a href={post.path}&gt;
                &lt;h1&gt;{post.title}&lt;/h1&gt;
              &lt;/a&gt;
              &lt;p&gt;{post.excerpt}&lt;/p&gt;
              &lt;Metadata tags={post.tags} date={post.date} timeToRead={post.timeToRead} /&gt;
              &lt;hr /&gt;
            &lt;/section&gt;
          )
        })}
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default PostListing
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 09:15:22 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
