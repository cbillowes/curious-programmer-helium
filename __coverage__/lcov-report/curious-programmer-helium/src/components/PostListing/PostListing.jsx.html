<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for curious-programmer-helium/src/components/PostListing/PostListing.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">curious-programmer-helium/src/components/PostListing</a> PostListing.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react"
import Metadata from "../Metadata/Post/Metadata"
import "./PostListing.scss"
&nbsp;
class PostListing extends React.Component {
  getPostList() {
    const postList = []
    this.props.postEdges.forEach(postEdge =&gt; {
      postList.push({
        path: postEdge.node.fields.slug,
        tags: postEdge.node.frontmatter.tags,
        cover: postEdge.node.frontmatter.cover,
        title: postEdge.node.frontmatter.title,
        date: postEdge.node.fields.date,
        excerpt: postEdge.node.excerpt,
        timeToRead: postEdge.node.timeToRead
      })
    })
    return postList
  }
  render() {
    const list = this.getPostList()
    return (
      &lt;div className="posts"&gt;
        {list.map(post =&gt; {
          return (
            &lt;section key={`post-${post.path}`}&gt;
              &lt;a href={post.path}&gt;
                &lt;h1&gt;{post.title}&lt;/h1&gt;
              &lt;/a&gt;
              &lt;p&gt;{post.excerpt}&lt;/p&gt;
              &lt;Metadata tags={post.tags} date={post.date} timeToRead={post.timeToRead} /&gt;
              &lt;hr /&gt;
            &lt;/section&gt;
          )
        })}
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default PostListing
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jan 05 2019 10:01:49 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
