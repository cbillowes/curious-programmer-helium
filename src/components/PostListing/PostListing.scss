@import "../../layout/layout.scss";

.posts {
  section {
    padding: 0em 2.5em;
    max-width: $container-width;
    margin: 0 auto;

    &:first-child {
      padding: 0em 2.5em;
      text-align: center;

      &::after {
        content: "";
        position: relative;
      }

      h1 {
        font-size: 3em;
        color: $primary;

        @include small-media() {
          font-size: 2em;
        }

      }

      .tags {
        .tag,
        &:visited {
          background-color: $primary;
          color: $background-accent;

          &:hover {
            background-color: $secondary;
          }
        }
      }
    }

    a {
      color: $foreground;

      &:hover {
        color: $secondary;
      }
    }
  }
}