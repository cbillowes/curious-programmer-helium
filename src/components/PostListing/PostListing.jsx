import React from "react"
import Metadata from "../Metadata/Post/Metadata"
import "./PostListing.scss"

class PostListing extends React.Component {
  getPostList() {
    const postList = []
    this.props.postEdges.forEach((postEdge) => {
      postList.push({
        path: postEdge.node.fields.slug,
        tags: postEdge.node.frontmatter.tags,
        cover: postEdge.node.frontmatter.cover,
        title: postEdge.node.frontmatter.title,
        date: postEdge.node.fields.date,
        excerpt: postEdge.node.excerpt,
        timeToRead: postEdge.node.timeToRead
      })
    })
    return postList
  }
  render() {
    const list = this.getPostList()
    return (
      <div className="posts">
        {list.map((post) => {
          return (
            <section key={`post-${post.path}`}>
              <a href={post.path}>
                <h1>{post.title}</h1>
              </a>
              <p>{post.excerpt}</p>
              <Metadata tags={post.tags} date={post.date} timeToRead={post.timeToRead} />
              <hr />
            </section>
          )
        })}
      </div>
    )
  }
}

export default PostListing
