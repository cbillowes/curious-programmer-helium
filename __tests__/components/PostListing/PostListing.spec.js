import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import PostListing from "../../../src/components/PostListing/PostListing"

const postEdges = {
  edges: [{
    node: {
      excerpt: "My first simple write up.",
      timeToRead: 20,
      fields: {
        slug: "/blog/getting-started-with-datomic-part-1",
        date: new Date()
      },
      frontmatter: {
        tags: ["technical", "clojure", "datomic"],
        title: "Getting started with Datomic part 1",
        cover: "https://picsum.photos/#/157"
      }
    }
  },
  {
    node: {
      excerpt: "My second simple write up.",
      timeToRead: 20,
      fields: {
        slug: "/blog/getting-started-with-datomic",
        date: new Date()
      },
      frontmatter: {
        tags: ["technical", "clojure", "datomic"],
        title: "Getting started with Datomic part 2",
        cover: "https://picsum.photos/#/564"
      }
    }
  }, {
    node: {
      excerpt: "A duplicate post.",
      timeToRead: 20,
      fields: {
        slug: "/blog/getting-started-with-datomic-part-2",
        date: new Date()
      },
      frontmatter: {
        tags: ["technical", "clojure", "datomic"],
        title: "Getting started with Datomic part 2",
        cover: "https://picsum.photos/#/564"
      }
    }
  }]
}

describe("a post listing component", () => {
  it("should render", () => {
    const element = <PostListing postEdges={postEdges.edges} />
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render container", () => {
    const component = mount(<PostListing postEdges={postEdges.edges} />)
    expect(component.find(".posts").length).toEqual(1)
  })

  it("should render all attributes", () => {
    const component = mount(<PostListing postEdges={postEdges.edges} />)
    postEdges.edges.map(edge => {
      expect(component.html()).toContain(`<p>${edge.node.excerpt}</p>`)
      expect(component.html()).toContain(`<h1>${edge.node.frontmatter.title}</h1>`)
      edge.node.frontmatter.tags.map(tag => {
        expect(component.html()).toContain(`>${tag}</`)
      })
      expect(component.html()).toContain(`<a href="${edge.node.fields.slug}"`)
      expect(component.html()).toContain(`a few seconds ago</time>`)
    })
  })

  it("should render all posts", () => {
    let duplicatedEdges = { edges: [] }
    for (let i = 0; i < 4; i++) {
      duplicatedEdges.edges.push({
        node: {
          fields: {
            slug: "/blog/getting-started-with-datomic/" + i
          },
          frontmatter: {}
        }
      })
    }
    const component = mount(<PostListing postEdges={duplicatedEdges.edges} />)
    expect(component.find("a").length).toEqual(duplicatedEdges.edges.length)
  })
})