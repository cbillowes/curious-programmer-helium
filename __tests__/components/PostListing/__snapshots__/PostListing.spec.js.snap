// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a post listing component should render 1`] = `
<div
  className="posts"
>
  <section>
    <a
      href="/blog/getting-started-with-datomic-part-1"
    >
      <h1>
        Getting started with Datomic part 1
      </h1>
    </a>
    <p>
      My first simple write up.
    </p>
    <div
      className="post-metadata"
    >
      <p>
        posted 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          a few seconds ago
        </time>
         
         
        on 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          Sat, 5 January 2019
        </time>
         
         
        (Est. 
        20
         minute read)
      </p>
      <ul
        className="tags"
      >
        <li>
          <a
            className="tag"
            href="/tags/technical"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            technical
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/clojure"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            clojure
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/datomic"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            datomic
          </a>
        </li>
      </ul>
    </div>
    <hr />
  </section>
  <section>
    <a
      href="/blog/getting-started-with-datomic"
    >
      <h1>
        Getting started with Datomic part 2
      </h1>
    </a>
    <p>
      My second simple write up.
    </p>
    <div
      className="post-metadata"
    >
      <p>
        posted 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          a few seconds ago
        </time>
         
         
        on 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          Sat, 5 January 2019
        </time>
         
         
        (Est. 
        20
         minute read)
      </p>
      <ul
        className="tags"
      >
        <li>
          <a
            className="tag"
            href="/tags/technical"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            technical
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/clojure"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            clojure
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/datomic"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            datomic
          </a>
        </li>
      </ul>
    </div>
    <hr />
  </section>
  <section>
    <a
      href="/blog/getting-started-with-datomic-part-2"
    >
      <h1>
        Getting started with Datomic part 2
      </h1>
    </a>
    <p>
      A duplicate post.
    </p>
    <div
      className="post-metadata"
    >
      <p>
        posted 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          a few seconds ago
        </time>
         
         
        on 
         
        <time
          dateTime={"2019-01-05T06:58:02.460Z"}
        >
          Sat, 5 January 2019
        </time>
         
         
        (Est. 
        20
         minute read)
      </p>
      <ul
        className="tags"
      >
        <li>
          <a
            className="tag"
            href="/tags/technical"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            technical
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/clojure"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            clojure
          </a>
        </li>
        <li>
          <a
            className="tag"
            href="/tags/datomic"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            datomic
          </a>
        </li>
      </ul>
    </div>
    <hr />
  </section>
</div>
`;
